### ISSUE TITLE
[Bug] "Error: invalid host format" when adding database host

### ISSUE THREAD

**Issue by bikerdave** [hestiacp/hestiacp#4522]
### Describe the bug

The URL validator when adding a database host using the CLI `v-add-database-host` only allows endpoints up to 64 chars. On cloud platforms the autogenerated hosts can often have more than this.

I reported this years ago on [VestaCP(1916)](https://github.com/outroll/vesta/issues/1916) but as this validator is used in other places didn't want to open a PR that might have adverse effects on other functionality.

### Tell us how to replicate the bug

`v-add-database-host mysql long-url-generated-by-cloud-platform-and-is-more-than-64-chars-long.example.com admin password`

### Which components are affected by this bug?

Database (MariaDB, MySQL, PostgreSQL)

### Hestia Control Panel Version

1.18.11

### Operating system

Ubuntu *.04

**Comment by jaapmarcus** [comment]
I am sure the check is  the same:
https://github.com/hestiacp/hestiacp/blob/dbc285b63f2d8934f625b95d3f2a77ffaaeab320/func/main.sh#L1205-L1209

https://github.com/outroll/vesta/blob/9de1d46bb6657281bf7cfd88ef28b0e744e77619/func/main.sh#L795-L800

**Comment by bikerdave** [comment]
Yeah, it was never upped on VestaCP so is the same with HestiaCP.

I guess I was querying if it would be safe to increase this limit so it's possible to add hosts that are longer than 64 chars?

**Comment by Skamasle** [comment]
That is RFC recommendations, but may can be changed.

I think this can be added to hestia.conf

**Comment by bikerdave** [comment]
@Skamasle Good to know the rationale behind the limit, thanks!

**Comment by Skamasle** [comment]
Hi

I think the rule is miss interpreted

RFC 2.3.4 say this:

```
[2.3.4](https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.4). Size limits

Various objects and parameters in the DNS have size limits.  They are
listed below.  Some could be easily changed, others are more
fundamental.

labels          63 octets or less

names           255 octets or less
```

So max.charactets are 255 in total, and 63 per label.

So, each label is content separated from each dot.

Subdomain-with-64chars.other-subdomain-with-64-characters.total255chars.com

We need modify that test, total 255 chars and maximum of 64 per label.

Later I will try code some different verification to apply the that RFC

**Comment by Skamasle** [comment]
I dont know what exactly check this function, because this validate hosts like subdominio....domain.com

I make new function but may @jaapmarcus can check it, I dont know if is really needed a underscore here.

    # Problems I see, old function validate this:
    # domain..domain.com and this domain_domain.domain.com but not:
    # _domain.domain.com ( some DNS have this format
    # This new validate: domain_domain.com, _domain_domain.com and fix
    # Domain..domain.com, and check 253 total lengh and 63 for each label.
    # I recomend remove _ from this check, if hist just validate hostnames

```
is_object_format_valid() {
        if [[ "${#1}" -gt "253" ]] || [[ "$1" =~ \.\. ]]; then
                check_result "$E_INVALID" "invalid $2 format :: $1"
                exit 1
        fi
        if ! [[ "$1" =~  ^[-._[:alnum:]]+$ ]]; then
                check_result "$E_INVALID" "invalid $2 format :: $1"
                exit 1
        fi
        labels=($(echo "$1" | tr '.' ' '))
        for label in "${labels[@]}"; do
        if [[ "${#label}" -gt "63" ]] || [[ "$label" =~ ^- || "$label" =~ -$ ]]; then
                check_result "$E_INVALID" "invalid $2 format :: $1"
                break
        fi

        done
}
```
